lib/runtime/wazero/imports.go

```diff
@@ -0,0 +1,2247 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package wazero_runtime
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"reflect"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
+	"github.com/ChainSafe/gossamer/lib/crypto/secp256k1"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/lib/trie/proof"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/tetratelabs/wazero/api"
+)
+
+var (
+	logger = log.NewFromGlobal(
+		log.AddContext("pkg", "runtime"),
+		log.AddContext("module", "wazero"),
+	)
+)
+
+const (
+	validateSignatureFail = "failed to validate signature"
+)
+
+// toPointerSize converts an uint32 pointer and uint32 size
+// to an int64 pointer size.
+func newPointerSize(ptr, size uint32) (pointerSize uint64) {
+	return uint64(ptr) | (uint64(size) << 32)
+}
+
+// splitPointerSize converts a 64bit pointer size to an
+// uint32 pointer and a uint32 size.
+func splitPointerSize(pointerSize uint64) (ptr, size uint32) {
```

Comment:
previously this took in an int64 and cast to a pointer and size represented as uint32's, why did this functionality change?

Did the usage just change with using wazero? If this is intended behavior we should update the comments for these two functions

Reply:
I'm representing a `pointerSize` as uint64.  Whether or not it actually will use the entire bounds of a `uint64` is another question, but technically by using `uint64` we have more possible values for the combined `pointerSize`.

Reply:
yeah I think some of the new code was mixed with old comments which threw me